lms/templates/admin/instance.canvas_api_details.html.jinja2

```diff
@@ -0,0 +1,18 @@
+{% import "macros.jinja" as macros %}
```

Comment:
Having an extra template for this form is IMO the right SRE way of doing this. 

It adds a bit more boilerplate but trying to have anything resembling a dialog would just make it more complex UI and code wise.